src/utils/with-middleware.ts

```diff
@@ -9,10 +9,10 @@ export const withMiddleware = (
 ): SWRHook => {
   return <Data = any, Error = any>(
     ...args:
-      | readonly [Key]
-      | readonly [Key, Fetcher<Data> | null]
-      | readonly [Key, SWRConfiguration | undefined]
-      | readonly [Key, Fetcher<Data> | null, SWRConfiguration | undefined]
```

Comment:
why removing `readonly`?

Reply:
This was used to make tuple inference works before TS 4.0. 
https://github.com/vercel/swr/pull/741/files#diff-34add1ba5917e182ff385ebefbcff5dd9b012723f395e89d77548c26c9f8735cR48-R52

Since we have upgraded the typescript version we don't need it any more